plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -340,6 +313,62 @@ public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHand
         }
     }
 
+    private Optional<JdbcColumnHandle> getColumn(ConnectorSession session, SchemaTableName schemaTableName, Connection connection, ResultSet resultSet)
+            throws SQLException
+    {
+        String columnName = resultSet.getString("COLUMN_NAME");
+        try {
+            return getColumn(session, schemaTableName, connection, resultSet, columnName);
+        }
+        catch (SQLException | RuntimeException e) {
```

Comment:
> When resolution of the column fails (e.g. bug in the jdbc driver) there
is not enough information to track what column has failed, e.g.

That could be true for mostly any method that is using JDBC. `toColumnMapping` is not really very special here, the difference is that it just happened to you recently (which may suggest that it fails more frequently than others, but that may not be true as we don't have enough data).

Anyway I think the thing you try to implement is already implemented in proper way for all method for `JdbcClient` in `io.trino.plugin.jdbc.JdbcDiagnosticModule#createJdbcClientWithStats`

Reply:
I'm not sure if I follow how this can help to understand what exact table column has problem. 
Stats is more about distribution of all problems jdbc client faces.

Reply:
> Stats is more about distribution of all problems jdbc client faces.

I don't mean stats, see `io.trino.plugin.base.util.LoggingInvocationHandler` usage

Reply:
This is very nice feature, however `toColumnMapping` does not receive column name as a parameter. Trino users have tables with a lot of columns. Forth-back moves to investigate exact problem is tedious.

Reply:
> does not receive column name as a parameter. 

Maybe we could add it there then?